recipes/automated-testing.md

```diff
@@ -0,0 +1,239 @@
+# Automated Testing
+
+## Testing best practices
+
+Writing tests can be hard because there are a lot of things that can be tested.
+
+Starting out can be overwhelming. But since writing tests is easy, we can write
+a lot of them in no time. Don't do this. Do not focus on code coverage number as
+it can lead to false sense of security. Remember that code with 100% test coverage
+can still have bugs.
+
+Focus on test quality and test performance. Make sure the test is not asserting
+unimportant things. Make sure the test is as quick as possible. Quick tests will
+be run often. Running tests often means more early bug detection which means less
+production errors.
+
+---
+
+Deal with flaky tests immediately. Flaky tests ruin test suite confidence. A failed
+test should raise alarm immediately. If the test suite contains flaky tests, disable
+them and refactor as soon as possible.
```

Comment:
I'll comment on this part but it relates to the whole document. I feel there is a lack of real-world examples, and scenarios and that is why in the end this document may not bring much value to those who will be writing tests in the future. On the example of the flaky tests, things worth mentioning that give a better understanding and tackle real-world issues,
- what are typical causes, e.g. concurrency, external dependencies, ordering, timing, etc.
- what are the signs, e.g. inconsistent results between runs, reliance on ordering/timing, etc.
- how to detect, e.g. run tests in parallel, run in different environments, review test results, etc.
- how to prevent, e.g. isolation, concurrency handling, setup and cleanup routines, predictable test data generation, etc.
- adding some code examples of actual flaky tests for even easier understanding

Reply:
@droguljic welcome back! This document is work in progress and the idea is to add real-world examples here, but at a later point.
First, I want to nail this "abstract" part for the first version and then work with the team to add examples.
Please check the formatting and the overall tone of this doc. If we want to "refactor" this, this is the time to do it while it still hasn't grown too much.

Reply:
Thanks! I'll more closley check the formatting and overall doc tone. Still, some parts, like the previously mentioned flaky tests sections seem more of a placeholder even for the abstract version. I think we should make some sections more descriptive and informative for the first pass also.